### PR TITLE
[macOS] Adding Swig

### PR DESCRIPTION
**Pull request by kassannk**
# Description
PR with swig 4.0.2 installation. Script tests have been added, also packages are installed through the homebrew package manager.

#### Related issue:
https://github.com/actions/virtual-environments/issues/3342

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 1]
```diff
@@ -487,6 +487,11 @@ function Get-PowershellVersion {
     return $powershellVersion
 }
 
+function Get-SwigVersion {
+    $swigVersion = Run-Command "swig -version"
+    return "Swig $swigVersion"
```
Comment (miketimofeev) [inline images/macos/software-report/SoftwareReport.Common.psm1:492]: should it be either `swig` as it's for windows or `SWIG` as it's in the official docs?

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 2]
```diff
@@ -487,6 +487,11 @@ function Get-PowershellVersion {
     return $powershellVersion
 }
 
+function Get-SwigVersion {
+    $swigVersion = Run-Command "swig -version"
```
Comment (miketimofeev) [inline images/macos/software-report/SoftwareReport.Common.psm1:491]: This returns not the version, but the full output
image

File: images/macos/tests/Common.Tests.ps1 [hunk 3]
```diff
@@ -85,4 +85,10 @@ Describe "VSMac" {
         $vsPath | Should -Exist
         $vstoolPath | Should -Exist
     }
-}
\ No newline at end of file
+}
+
+Describe "Swig" {
```
Comment (miketimofeev) [inline images/macos/tests/Common.Tests.ps1:90]: we should either add Swig to macOS 10.14 or add a skip condition to the test.

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 1 above)
Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1:492]: Fixed. Swig - so as not to break the code style

File: images/macos/tests/Common.Tests.ps1 (hunk 3 above)
Comment (kassannk) [reply images/macos/tests/Common.Tests.ps1:90]: added for macOS 10.14

File: images/macos/software-report/SoftwareReport.Common.psm1 (hunk 2 above)
Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1]: Fixed

Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1]: fixed

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 4]
```diff
@@ -487,6 +487,11 @@ function Get-PowershellVersion {
     return $powershellVersion
 }
 
+function Get-SwigVersion {
+    $swigVersion = Run-Command "swig -version" | Take-Part -Part 3
```
Comment (AlenaSviridenko) [inline images/macos/software-report/SoftwareReport.Common.psm1:491]: This is the output of this command, output is incorrect
image

Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1:491]: ok

Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1]: fixed

Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1]: > This is the output of this command, output is incorrect
> image

fixed

Comment (kassannk) [reply images/macos/software-report/SoftwareReport.Common.psm1]: > This is the output of this command, output is incorrect
> image

fixed

**Review by dmitry-shibanov — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

**Pull request merged**